crates/ruff_dev/src/generate_docs.rs

```diff
@@ -49,3 +49,32 @@ pub fn main(args: &Args) -> Result<()> {
     }
     Ok(())
 }
+
+fn process_documentation(documentation: &str, out: &mut String) {
+    let mut in_options = false;
+    let mut after = String::new();
+
+    for line in documentation.split_inclusive('\n') {
+        if line.starts_with("## ") {
+            in_options = line == "## Options\n";
+        } else if in_options {
+            if let Some(rest) = line.strip_prefix("* `") {
+                // TODO(martin): Assert that the option actually exists
+
+                let option = rest.trim_end().trim_end_matches('`');
+                let anchor = option.rsplit('.').next().unwrap();
+                out.push_str(&format!("* [`{option}`]\n"));
+                after.push_str(&format!(
+                    "[`{option}`]: https://github.com/charliermarsh/ruff#{anchor}"
```

Comment:
Nit: Can we link to the docs?

Reply:
Sure :) Changed it to `../../settings#{anchor}` ... which *should* work ... I am not certain since I don't know how to build mkdocs locally, see #2802.